extensions/resteasy-reactive/rest-client-reactive/deployment/src/main/java/io/quarkus/rest/client/reactive/deployment/RestClientReactiveProcessor.java

```diff
@@ -151,6 +160,65 @@ void registerHeaderFactoryBeans(CombinedIndexBuildItem index,
         }
     }
 
+    /**
+     * Creates an implementation of `AnnotationRegisteredProviders` class with a constructor that
+     * puts all the providers registered by the @RegisterProvider annotation in a
+     * map using the {@link AnnotationRegisteredProviders#addProviders(String, Map)} method
+     *
+     * @param indexBuildItem index
+     * @param generatedBeans build producer for generated beans
+     */
+    @BuildStep
+    void registerProvidersFromAnnotations(CombinedIndexBuildItem indexBuildItem,
```

Comment:
I am fairly certain that the generated class can be replaced by a synthetic bean which would make maintenance a lot easier

Reply:
I'll see if I can do that

Reply:
I created a version with synthetic bean: https://github.com/quarkusio/quarkus/pull/16815
It has its own drawbacks: more code and has drawbacks like usage of `Class.forName` and registering classes for reflection. I have a slight preference on this version but I don't mind if we merge the other too.

Reply:
I prefer #16815, but it's up to you as this is your baby :)

Reply:
okay, let's go with this one then :)

If it's okay now, can you approve?

Reply:
As you wish :)(